src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -27,36 +27,57 @@ class CreateZuoraSubscription(servicesProvider: ServiceProvider = ServiceProvide
     requestInfo: RequestInfo,
     context: Context,
     services: Services
-  ): FutureHandlerResult = for {
-    identityId <- Future.fromTry(IdentityId(state.user.id))
-    maybeDomainSubscription <- GetRecurringSubscription(services.zuoraService, state.requestId, identityId, state.product.billingPeriod)
-    thankYouState <- maybeDomainSubscription match {
-      case Some(domainSubscription) => skipSubscribe(state, requestInfo, domainSubscription)
-      case None => subscribe(state, requestInfo, services)
-    }
-  } yield thankYouState
+  ): FutureHandlerResult = {
+    val subscribeItem = buildSubscribeItem(state, services.promotionService)
+    for {
+      identityId <- Future.fromTry(IdentityId(state.user.id))
+      maybeDomainSubscription <- GetRecurringSubscription(services.zuoraService, state.requestId, identityId, state.product.billingPeriod)
+      previewPaymentSchedule <- PreviewPaymentSchedule(subscribeItem, state.product.billingPeriod, services, checkSingleResponse)
+      thankYouState <- maybeDomainSubscription match {
+        case Some(domainSubscription) => skipSubscribe(state, requestInfo, domainSubscription)
+        case None => subscribe(state, subscribeItem, previewPaymentSchedule, requestInfo, services)
+      }
+    } yield thankYouState
+  }
 
   def skipSubscribe(state: CreateZuoraSubscriptionState, requestInfo: RequestInfo, subscription: DomainSubscription): FutureHandlerResult = {
     val message = "Skipping subscribe for user because they are already an active contributor"
     SafeLogger.info(message)
-    FutureHandlerResult(getEmailState(state, subscription.accountNumber, subscription.subscriptionNumber), requestInfo.appendMessage(message))
+    FutureHandlerResult(
+      getEmailState(state, subscription.accountNumber, subscription.subscriptionNumber, PaymentSchedule(Nil)), requestInfo.appendMessage(message)
```

Comment:
so if the lambda retries we will lose the payment schedule.  is that alright?  I don't think the lambda should returns something different if it retries.

Reply:
Very good point, I'll update this.